From 7da3c66c0d34df23fa126284c785f3be36134880 Mon Sep 17 00:00:00 2001
From: Julien Jerphanion <git@jjerphan.xyz>
Date: Wed, 26 Mar 2025 14:15:10 +0100
Subject: [PATCH] Add 'c++20' as an option for `MSVC_LANGUAGE_VERSION`

Signed-off-by: Julien Jerphanion <git@jjerphan.xyz>
---
 CMake/FollyCompilerMSVC.cmake | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/CMake/FollyCompilerMSVC.cmake b/CMake/FollyCompilerMSVC.cmake
index ad03039aa..af3f499b2 100644
--- a/CMake/FollyCompilerMSVC.cmake
+++ b/CMake/FollyCompilerMSVC.cmake
@@ -39,11 +39,12 @@ if (NOT MSVC_FAVORED_ARCHITECTURE STREQUAL "blend" AND NOT MSVC_FAVORED_ARCHITEC
   message(FATAL_ERROR "MSVC_FAVORED_ARCHITECTURE must be set to one of exactly, 'blend', 'AMD64', 'INTEL64', or 'ATOM'! Got '${MSVC_FAVORED_ARCHITECTURE}' instead!")
 endif()
 
-set(MSVC_LANGUAGE_VERSION "c++17" CACHE STRING "One of 'c++17', or 'c++latest'. This determines which version of C++ to compile as.")
+set(MSVC_LANGUAGE_VERSION "c++17" CACHE STRING "One of 'c++17', 'c++20' or 'c++latest'. This determines which version of C++ to compile as.")
 set_property(
   CACHE MSVC_LANGUAGE_VERSION
   PROPERTY STRINGS
     "c++17"
+    "c++20"
     "c++latest"
 )
 
-- 
2.48.1

