From acf5d8a13151d21783010a47f4f0281c0d0bf9ce Mon Sep 17 00:00:00 2001
From: "Uwe L. Korn" <uwe.korn@quantco.com>
Date: Mon, 31 Oct 2022 10:14:07 +0100
Subject: [PATCH] Use CMAKE_SYSTEM_PROCESSOR instead of
 CMAKE_LIBRARY_ARCHITECTURE

---
 CMakeLists.txt | 16 ++++++++--------
 1 file changed, 8 insertions(+), 8 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 9a4f762..b4b8093 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -76,13 +76,13 @@ fb_activate_static_library_option()
 set(CMAKE_CXX_STANDARD 17)
 set(CMAKE_CXX_STANDARD_REQUIRED ON)
 
-if("${CMAKE_LIBRARY_ARCHITECTURE}" STREQUAL "")
-  # CMAKE_LIBRARY_ARCHITECTURE is not always set, so we have to assume
+if("${CMAKE_SYSTEM_PROCESSOR}" STREQUAL "")
+  # CMAKE_SYSTEM_PROCESSOR is not always set, so we have to assume
   # arch might be x86_64
-  message(WARNING "CMAKE_LIBRARY_ARCHITECTURE not set, assuming x86_64")
+  message(WARNING "CMAKE_SYSTEM_PROCESSOR not set, assuming x86_64")
   set(IS_X86_64_ARCH ON)
 else()
-  string(FIND "${CMAKE_LIBRARY_ARCHITECTURE}" "x86_64" IS_X86_64_ARCH)
+  string(FIND "${CMAKE_SYSTEM_PROCESSOR}" "x86_64" IS_X86_64_ARCH)
   if(IS_X86_64_ARCH STREQUAL "-1")
     set(IS_X86_64_ARCH OFF)
   else()
@@ -244,14 +244,14 @@ endif()
 if (NOT(${IS_X86_64_ARCH}))
   message(
     STATUS
-    "arch ${CMAKE_LIBRARY_ARCHITECTURE} does not match x86_64, "
+    "arch ${CMAKE_SYSTEM_PROCESSOR} does not match x86_64, "
     "skipping building SSE4.2 version of base64"
   )
   list(REMOVE_ITEM files ${FOLLY_DIR}/detail/base64_detail/Base64_SSE4_2.cpp)
 else()
   message(
     STATUS
-    "arch ${CMAKE_LIBRARY_ARCHITECTURE} matches x86_64, "
+    "arch ${CMAKE_SYSTEM_PROCESSOR} matches x86_64, "
     "building SSE4.2 version of base64"
   )
   # MSVC does not have a way to enable just sse4.2, only avx.
@@ -270,13 +270,13 @@ if (${LIBSODIUM_FOUND})
   if (NOT(${IS_X86_64_ARCH}))
     message(
       STATUS
-      "arch ${CMAKE_LIBRARY_ARCHITECTURE} does not match x86_64, "
+      "arch ${CMAKE_SYSTEM_PROCESSOR} does not match x86_64, "
       "skipping setting SSE2/AVX2 compile flags for LtHash SIMD code"
     )
   else()
     message(
       STATUS
-      "arch ${CMAKE_LIBRARY_ARCHITECTURE} matches x86_64, "
+      "arch ${CMAKE_SYSTEM_PROCESSOR} matches x86_64, "
       "setting SSE2/AVX2 compile flags for LtHash SIMD code"
     )
     set_source_files_properties(
-- 
2.37.0 (Apple Git-136)

