From 57fee4739d9db241b58ed251129fb1ec8bea0d6a Mon Sep 17 00:00:00 2001
From: Uwe Korn <uwelk@xhochy.com>
Date: Fri, 5 Jul 2024 09:15:56 +0000
Subject: [PATCH] Add aarch64-based memcpy files to CMake build

---
 CMakeLists.txt | 22 ++++++++++++++++++++++
 1 file changed, 22 insertions(+)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 81c6d5a..cd9a043 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -85,6 +85,12 @@ else()
   set(IS_X86_64_ARCH FALSE)
 endif()
 
+IF(NOT DEFINED IS_AARCH64_ARCH AND ${CMAKE_SYSTEM_PROCESSOR} MATCHES "aarch64")
+  set(IS_AARCH64_ARCH TRUE)
+else()
+  set(IS_AARCH64_ARCH FALSE)
+endif()
+
 if(CMAKE_SYSTEM_NAME STREQUAL "Windows")
   # Check target architecture
   if (NOT CMAKE_SIZEOF_VOID_P EQUAL 8)
@@ -370,6 +376,22 @@ if (IS_X86_64_ARCH AND NOT MSVC)
   list(APPEND folly_base_files
     ${FOLLY_DIR}/memcpy.S
   )
+elseif(IS_AARCH64_ARCH)
+  set_property(
+    SOURCE
+    ${FOLLY_DIR}/external/aor/memcpy-armv8.S
+    ${FOLLY_DIR}/external/aor/memcpy_sve.S
+    ${FOLLY_DIR}/external/aor/memcpy-advsimd.S
+    ${FOLLY_DIR}/external/aor/memset-advsimd.S
+    APPEND PROPERTY COMPILE_OPTIONS "-x" "assembler-with-cpp"
+  )
+  list(APPEND folly_base_files
+    ${FOLLY_DIR}/external/aor/memcpy-armv8.S
+    ${FOLLY_DIR}/external/aor/memcpy_sve.S
+    ${FOLLY_DIR}/external/aor/memcpy-advsimd.S
+    ${FOLLY_DIR}/external/aor/memset-advsimd.S
+  )
+
 endif()
 
 add_library(folly_base OBJECT
